test/GithubApi.Repositories.test.js

```diff
@@ -0,0 +1,51 @@
+const { StatusCodes } = require('http-status-codes');
+const { expect } = require('chai');
+const axios = require('axios');
+const chai = require('chai');
+const chaiSubset = require('chai-subset');
+
+chai.use(chaiSubset);
+
+const urlBase = 'https://api.github.com';
+const githubUserName = 'aperdomob';
+const company = 'Perficient Latam';
+const name = 'Alejandro Perdomo';
+const repoName = 'jasmine-json-report';
+const desc = 'A Simple Jasmine JSON Report';
+
+describe('Github Api Test', () => {
+  it('GET method consumption', async () => {
+    const response = await axios.get(`${urlBase}/users/${githubUserName}`);
+    expect(response.status).to.equal(StatusCodes.OK);
+    expect(response.data.login).to.equal(githubUserName);
+    expect(response.data.company).to.equal(company);
+    expect(response.data.name).to.equal(name);
+  });
+  it('GET method for repos', async () => {
+    const repos = await axios.get(`${urlBase}/users/${githubUserName}/repos`);
```

Comment:
In the response of the previous test, you can get this URL:
![image](https://user-images.githubusercontent.com/10209180/176709019-ef048b00-0d63-4bec-a195-8483ff5ac1ee.png)

You have two options:

1. Define a global variable, that is assigned in the previous test, as the value change after is declared, you should declare it as `let` before the `describe`. This approach has the inconvenience that the tests will depend on one on another, but it is ok for this exercise.
2. The other option is to use the [hooks](https://mochajs.org/#hooks), to initialize the required data in tests (`it`). And nest the describes to reuse the data. One example of the structure of the tests with this approach is this:

```
describe('Github Api Test', () => {
  let userResponse;
  
  before(async () => {
    userResponse = await axios.get(`${urlBase}/users/${githubUserName}`);
  });
  
  it('GET method consumption', () => {
    expect(userResponse.status).to.equal(StatusCodes.OK);
    expect(userResponse.data.login).to.equal(githubUserName);
    expect(userResponse.data.company).to.equal(company);
    expect(userResponse.data.name).to.equal(name);
  });
  
  describe('Using hypermedia', () => {
    let jasmineRepo;
    before(async () => {
      const repositoriesResponse = await axios.get(`${userResponse.repos_url}`);
      jasmineRepo = repositoriesResponse.data.find(
        (repo) => repo.name === repoName);
    });
    
    it('to GET method for repos', () => {
      expect(jasmine.name).to.equal(repoName);
      expect(jasmine.private).to.equal(false);
      expect(jasmine.description).not.equal(desc);
    });
    
    it('Downloading by zip', async () => {
      // code to verify the zip download
    });
    
    describe('and to verify content', () => {
      // code to verify the readme and to download and verify the md5
    });
  });
});
```

Reply:
I'm editing the code and this error appears.
I'm not sure if this error is caused by a bad describe
![imagen](https://user-images.githubusercontent.com/36246464/180349185-4553ad6b-dbfa-4e79-adc7-286e8f81f00f.png)

Reply:
In line 23, you don't need an async since there is no calls to APIs through axios in that `it` section

Reply:
If you continue having issues, you can push the changes and we will try to find the root cause of the issue

Reply:
I found the issue, I was making a http request to an undefined url. I just change userResponse.data.repos_url instead userResponse.repos_url